content/actions/learn-github-actions/expressions.md

```diff
@@ -15,7 +15,7 @@ versions:
 
 You can use expressions to programmatically set environment variables in workflow files and access contexts. An expression can be any combination of literal values, references to a context, or functions. You can combine literals, context references, and functions using operators. For more information about contexts, see "[AUTOTITLE](/actions/learn-github-actions/contexts)."
 
-Expressions are commonly used with the conditional `if` keyword in a workflow file to determine whether a step should run. When an `if` conditional is `true`, the step will run.
+Expressions are commonly used with the conditional `if` keyword in a workflow file to determine whether a step should run. When an `if` conditional is `truthy` (in [JavaScript sense](https://developer.mozilla.org/en-US/docs/Glossary/Truthy)), the step will run. All values are `truthy` except for `false`, `0`, `-0`, `0n`, `""`, `''`, `null`, `undefined`, `NaN`. You can evaluate an `!!<expression>` (e.g., in your browser console) to check whether the `<expression>` is `truthy`. For example, `!!"false"` produces `true`, so `"false"` is `truthy`.
 
```

Comment:
This... isn't correct at all.

The workflow is parsed in a string context (except for numerals and booleans), so JavaScript truthiness is irrelevant. There's no way to define a true JavaScript `NaN` or `undefined` in YML, you end up with the strings `'NaN'` or `'undefined'`. Nearly any string will evaluate `true`.

In this workflow:
```yml
    strategy:
      matrix:
        java: [17]
        runme:
          - NaN
          - undefined
          - yes
          - no
          - false
          - 0
          - 1
          - -0

    steps:
      - name: true-path
        if: matrix.runme
        run: echo "${{ matrix.runme }} evaluates true"
      - name: false-path
        if: '! matrix.runme'
        run: echo "${{ matrix.runme }} evaluates false"
```

...The only `matrix.runme` value for which `false-path` will be executed are `false` and `0`. (And `-0`, technically, but that's only because it's interpreted as `0`. There's no distinction in YAML.) All of the others will evaluate `true`.

Reply:
Oh, sorry, `null` is also a valid YAML primitive, and it will also evaluate `false`.

Honestly, I think this is pretty well explained in [the Literals section of the Expressions docs](https://docs.github.com/en/actions/learn-github-actions/expressions#literals).

Reply:
1. It's weird (for me) that `'! matrix.runme'` evaluates to a condition. I'd expect that only `! matrix.runme` or `${{ matrix.runme }}` evaluates to a condition.
2. I ran the suggested job ([link](https://github.com/deemp/gists/actions/runs/7665774871)). I can confirm that the non-truthy values are `false`, `0`, `-0`, `""`, `''`, `null`. I can't see where these values are listed in [the Literals section](https://docs.github.com/en/actions/learn-github-actions/expressions#operators). What do you think about just extending the `boolean` docs?

Data type | Literal value
---------------|-------------------
`boolean`  | falsy (`false`, `0`, `-0`, `""`, `''`, `null`) or truthy (non-falsy)

Reply:
Updated the PR